packages/itwinui-react/scripts/build.mjs

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { execSync } from 'node:child_process';
+
+// https://swc.rs/docs/usage/cli
+const swcOptions = {
+  shared: [
+    '--strip-leading-paths',
+    '--no-swcrc',
+    '--ignore **/*.test.*,**/styles.js/*,**/styles*',
+  ].join(' '),
+
+  // https://swc.rs/docs/configuration/compilation
+  compilerOptions: [
+    '',
+    'jsc.parser.syntax=typescript',
+    'jsc.parser.tsx=true',
+    'jsc.transform.react.useBuiltins=true',
+    'jsc.target=es2020',
+    'jsc.minify.format.comments=false',
+    'jsc.externalHelpers=true',
+    'module.noInterop=true',
```

Comment:
The first four options are the same as what's in the `.swcrc` file used during dev. The most important one of them is `es2020`; changing this could be a breaking change.

`jsc.externalHelpers=true` is explained in https://github.com/iTwin/iTwinUI/pull/2100#discussion_r1638295231.

`jsc.minify.format.comments=false` is mentioned in PR description. It's something that wasn't possible using just `tsc`, which means that our build output often contains useless comments like `eslint-disable` and `ts-ignore`.

[`module.noInterop=true`](https://swc.rs/docs/configuration/modules#nointerop) is the one I'm least sure about. I added it because I noticed a _lot_ of extra helpers in the CJS output, which felt overkill. I still need to research and test the potential impact of this.